### PR TITLE
docs: restructure and update TRG 5.01 and 5.05

### PR DESCRIPTION
**Pull request by almadigabor**
## Description

- Restructure **TRG501** and **TRG5.05**, most values file related info over
- Add examples and rules for chart values like ingress, dependencies, container

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: docs/release/trg-5/trg-5-05.md [hunk 1]
```diff
@@ -47,3 +49,55 @@ This might cause unexpected effects, as the Helm release could change without bu
...
+
+## Dependencies
+
+An example section for Postgresql dependency configuration:
+
+```yaml
+postgresql:
+  ## PostgreSQL chart configuration
+  enabled: true
+  ## Dependencies should be prefixed with the `nameOverride` and/or `fullnameOverride` properties
```
Comment (evegufy) [inline docs/release/trg-5/trg-5-05.md:92]: I'd refrain from recommending fullnameOverride(s) as especially in the case of umbrella charts unique labels (name) for dependencies are important and a fullnameOverride will lead to non-unique labels (at least if there are >1 fullnameOverride(s) in the umbrella).

**Review by evegufy — CHANGES_REQUESTED** [review]
Looks good but just some notes.

File: docs/release/trg-5/trg-5-01.md [hunk 2]
```diff
@@ -18,18 +19,14 @@ All Eclipse Tractus-X components **must** be installable via Helm and meet follo
 chart **has to meet** all of these criteria!
 
 - Helm chart **must be released**.
-- Appropriate versioning for `version` and `appVersion` has to be used in `Chart.yaml`
+- Appropriate versioning stratetgy must be used for `version` and `appVersion` in the `Chart.yaml` file
```
[the remark this answers is not in the document]
Comment (almadigabor) [reply docs/release/trg-5/trg-5-01.md]: Thanks, fixed the typo

File: docs/release/trg-5/trg-5-05.md [hunk 3]
```diff
@@ -2,32 +2,34 @@
...
 
-Best practices for Helm chart `values.yaml` file.
+Rules and best practices for Helm chart `values.yaml` file.
 
 ## Container images
 
+- Image tag **must** not be set by default, instead it should use the
+  `appVersion` property from the `Chart.yaml` file
+- `pullPolicy` **must** default to 'IfNotPresent'. If image tag is 'latest' it
+  should be changed to 'Always'
```
[the remark this answers is not in the document]
Comment (almadigabor) [reply docs/release/trg-5/trg-5-05.md]: Good point, moved it to the caution section with additional info.

File: docs/release/trg-5/trg-5-05.md (hunk 1 above)
Comment (almadigabor) [reply docs/release/trg-5/trg-5-05.md]: Point taken, I've removed the fullNameOverride from the comment.

File: docs/release/trg-5/trg-5-05.md [hunk 4]
```diff
@@ -36,14 +37,66 @@ image:
...
+  #  nginx.ingress.kubernetes.io/enable-cors: "true"
+  #  nginx.ingress.kubernetes.io/cors-allow-origin: "https://*.example.org"
+  tls:
+    - secretName: ""
+      hosts:
+        - ""
+  hosts:
+    - host: ""
+      paths:
+        - path: "/(.*)"
```
Comment (SebastianBezold) [inline docs/release/trg-5/trg-5-05.md:76]: Im not a 100%, but quite sure, that this regex/wildcard is not needed

**Review by FaGru3n — CHANGES_REQUESTED** [review]

File: docs/release/trg-5/trg-5-05.md [hunk 5]
```diff
@@ -2,32 +2,33 @@
...
 
 ## Description
 
-Best practices for Helm chart `values.yaml` file.
+Rules and best practices for Helm chart `values.yaml` file.
 
 ## Container images
 
+- Image tag **must not** be set by default, instead it should use the
+  `appVersion` property from the `Chart.yaml` file
```
Comment (evegufy) [inline docs/release/trg-5/trg-5-05.md:21]: I didn't noticed this at may first review and I wonder why define this in a such restrictive manner?
I mean it's good to default to the Charts.AppVersion if no image tag is defined but why not allow to set image tags as long as they are pointing to a versioned tag? Especially in the case of charts with multiple images this just causes unnecessary inflexibility.

**Review by evegufy — CHANGES_REQUESTED** [review]

File: docs/release/trg-5/trg-5-05.md [hunk 6]
```diff
@@ -36,14 +39,66 @@ image:
...
+## Dependencies
+
+An example section for Postgresql dependency configuration:
+
+```yaml
+postgresql:
+  ## PostgreSQL chart configuration
+  enabled: true
+  ## Dependencies should be prefixed with the `nameOverride` property
+  nameOverride: "app-postgresql"
```
Comment (evegufy) [inline docs/release/trg-5/trg-5-05.md:94]: Could you please expand on why setting a nameOverride is beneficial? I see the benefits when using the chart then in an umbrella scenario but why setting it in the product specific charts already I don't get. 
IMO the default naming behaviour of the Bitnami chart is sufficient and no nameOverride is needed.

**Review by evegufy — CHANGES_REQUESTED** [review]

File: docs/release/trg-5/trg-5-05.md (hunk 6 above)
Comment (evegufy) [reply docs/release/trg-5/trg-5-05.md:94]: and sorry that I didn't spot this at my last review

Comment (almadigabor) [reply docs/release/trg-5/trg-5-05.md]: made some adjustments here

**Review by FaGru3n — APPROVED** [review]
LGTM

**Pull request merged**